### PR TITLE
Refactored `genomics.py` and `metabolomics.py` by moving classes into individual files

### PR DESCRIPTION
**Pull request by hechth**
- Extracted BGC, GCF and MiBIGBGC classes from `genomics.py`
- Moved `genomics.py` and `genomics_utilities.py` into `genomics` subfolder together with extracted classes
- Extracted `MolecularFamily` class from `metabolomics.py`
- Extracted `SingletonFamily` class from `metabolomics.py`
- Extracted `Spectrum` class from `metabolomics.py`
- Moved all metabolomics related files into metabolomics subfolder.
- Added `__init__.py` files to new `genomics` and `metabolomics` folders

### CODE REVIEW

File: src/nplinker/metabolomics/load_gnps.py [hunk 1]
```diff
@@ -755,30 +414,10 @@ def load_clusterinfo_fbmn(strains, nodes_file, extra_nodes_file, md_table_file,
...
-    # above returns a list, create a dict indexed by spectrum_id to make
-    # the rest of the parsing a bit simpler
-    spec_dict = {spec.spectrum_id: spec for spec in spectra}
-
-    # add edges info to the spectra
-    load_edges(edges_file, spec_dict)
-
-    gnps_format = identify_gnps_format(nodes_file, quant_table_file
-                                       is not None)
+def load_gnps(strains, nodes_file, quant_table_file, metadata_table_file,
```
Comment (CunliangGeng) [inline src/nplinker/metabolomics/load_gnps.py:417]: The function name and return values are different with previous function, please keep them consistent

File: src/nplinker/metabolomics.py [hunk 2]
```diff
@@ -788,52 +427,14 @@ def load_dataset(strains,
...
+                                                nodes_file, spec_dict)
     else:
         logger.info('quantification table exists, new-style GNPS dataset')
-        _, unknown_strains = load_clusterinfo_fbmn(strains, nodes_file,
-                                                   quant_table_file,
-                                                   metadata_table_file,
-                                                   spec_dict,
-                                                   ext_metadata_parsing)
-
-    molfams = make_families(spectra)
```
Comment (CunliangGeng) [inline src/nplinker/metabolomics.py:801]: why removing this line?

File: src/nplinker/metabolomics/load_gnps.py [hunk 3]
```diff
@@ -788,52 +427,14 @@ def load_dataset(strains,
...
+        _, unknown_strains = _load_clusterinfo_fbmn(strains, nodes_file,
+                                                    quant_table_file,
+                                                    metadata_table_file,
+                                                    spec_dict,
+                                                    ext_metadata_parsing)
 
-    logger.debug('make_families: {} molams + {} singletons'.format(
-        fams, singles))
-    return families
+    return unknown_strains
```
Comment (CunliangGeng) [inline src/nplinker/metabolomics/load_gnps.py:440]: the return value is different with that of old function, why changing it? 
Did you check the code that calls this function?

File: src/nplinker/genomics/__init__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import logging
+
+
+logging.getLogger(__name__).addHandler(logging.NullHandler())
+
+__author__ = "Helge Hecht"
+__email__ = "h.hecht@esciencecenter.nl"
+__version__ = "1.3.2"
```
Comment (CunliangGeng) [inline src/nplinker/genomics/__init__.py:8]: Please remove this part. Let's keep these info only in the top __init__.py file.

File: src/nplinker/metabolomics/__init__.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import logging
+
+
+logging.getLogger(__name__).addHandler(logging.NullHandler())
+
+__author__ = "Helge Hecht"
+__email__ = "h.hecht@esciencecenter.nl"
+__version__ = "1.3.2"
```
Comment (CunliangGeng) [inline src/nplinker/metabolomics/__init__.py:8]: Please remove this part. Let's keep these info only in the top __init__.py file.

**Review by CunliangGeng — COMMENTED** [review]
Thanks for the refactoring. Changes look good, only two minor comments:
1. please rename new modules' name based on [pep8](https://realpython.com/python-pep8/#naming-styles) , e.g.  change BGC.py to bgc.py
2. we don't have test files currently, to make sure the changes work, please run a case locally (e.g. run the demo1 notebook)

**Comment by hechth** [comment]
The demo notebook works, so the import hierarchy is correct (it seems).

I'll see if I can implement further tests ad hoc.

I'll make the requested modifications :).

**Comment by CunliangGeng** [comment]
BTW, we have a [README for developer](https://github.com/NPLinker/nplinker/blob/main/README.dev.md) for this project, please have a look if not yet : -)

File: src/nplinker/genomics/__init__.py (hunk 4 above)
Comment (hechth) [reply src/nplinker/genomics/__init__.py]: Removed in https://github.com/NPLinker/nplinker/pull/78/commits/c2a10d7fc5231eae0bed5c2c97d5b5b057644243

File: src/nplinker/metabolomics/__init__.py (hunk 5 above)
Comment (hechth) [reply src/nplinker/metabolomics/__init__.py]: Removed in https://github.com/NPLinker/nplinker/pull/78/commits/c2a10d7fc5231eae0bed5c2c97d5b5b057644243

**Comment by hechth** [comment]
> Thanks for the refactoring. Changes look good, only two minor comments:
> 
> 1. please rename new modules' name based on [pep8](https://realpython.com/python-pep8/#naming-styles) , e.g.  change BGC.py to bgc.py
> 2. we don't have test files currently, to make sure the changes work, please run a case locally (e.g. run the demo1 notebook)

Renamed the files in https://github.com/NPLinker/nplinker/pull/78/commits/c410d6512722b2a8b92dcd10a86b48e4334e4dcc

File: conda/environment-dev.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+name: nplinker-dev
```
Comment (CunliangGeng) [inline conda/environment-dev.yaml:1]: Please remove this conda env file. 
If you like conda, use it only to create virtual env, and don't use conda packages (that's also what I'm doing). Let's install all packages from pip and the `install-nplinker-deps` command. The reason is that at the moment we don't want to solve conda conflicts for installing nplinker and its dependencies.

To create an dev env:
```
conda create -y --name nplinker-dev python=3.9
pip install -e .
install-nplinker-deps
```
I'll update README to make creating dev env clear.

File: src/nplinker/straincollection.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
+import csv
```
Comment (CunliangGeng) [inline src/nplinker/straincollection.py:1]: Rename the filename to `strain_collection.py`? to follow the style of e.g. `molecular_family.py`

**Review by CunliangGeng — APPROVED** [review]
Thanks, left you two minor comments, you can merge then after revising.

**Pull request merged**